src/massDistribution/Massdistribution.cpp

```diff
@@ -41,5 +41,106 @@ double DensityList::getNucleonDensity(const Vector3d &position) const {
 	return n;
 }
 
-} //namespace crpropa
+std::string DensityList::getDescription() {
+	std::stringstream ss; 
+	ss << "DensityList with " << DensityList.size() << " modules: \n";
+	for(int i = 0; i < DensityList.size(); i++){
+		ss << "density " << i + 1 << ": " << DensityList[i] -> getDescription();
+	}
+	
+	return ss.str();
+}
+
+// ----------- DensityGrid -----------------------------------------------------------------
+
+DensityGrid::DensityGrid(ref_ptr<Grid1f> grid, bool isForHI, bool isForHII, bool isForH2) : 
+	grid(grid), isForHI(isForHI), isForHII(isForHII), isForH2(isForH2) {
+		checkAndWarn();
+	}
+
+void DensityGrid::checkAndWarn() {
+	bool allDeactivated = (isForHI == false) & (isForHII == false) & (isForH2 == false);
```

Comment:
For **consistency**, you may want to use also `&&` as you do at another part of the code (or vice versa)